src/main/java/apartment/ui/Controller.java

```diff
@@ -0,0 +1,197 @@
+            }
+        } while (option != MainMenuOption.EXIT);
+    }
+
+    public void displayViewReservation() throws DataException {
+        view.displayHeader(MainMenuOption.VIEW_RESERVATION.getMessage());
+
+        String hostEmail = view.getUserStringInput("Host Email: ");
+
+        Host host = hostService.findHostFromEmail(hostEmail);
+
+        if (host == null) {
+            view.hostNotFound(hostEmail);
+            return;
+        }
+
+        List<Reservation> hostReservations = reservationService.getReservationForParticularHost(host);
+        view.printReservations(host, hostReservations);
+    }
+
+    public void displayCancelReservation() throws DataException {
+        view.displayHeader(MainMenuOption.CANCEL_RESERVATION.getMessage());
+        String guestEmail = view.getUserStringInput("Guest Email: ");
+        Guest guest = guestService.findGuestByEmail(guestEmail);
+        if (guest == null) {
+            view.guestNotFound(guestEmail);
+            return;
+        }
+        String hostEmail = view.getUserStringInput("Host Email: ");
+        Host host = hostService.findHostFromEmail(hostEmail);
+        if (host == null) {
+            view.hostNotFound(hostEmail);
+            return;
+        }
+
+
+        List<Reservation> reservations = reservationService.getReservationForParticularHost(host);
+
+        reservations = reservations.stream().filter(reservation -> reservation.getGuest().getEmail().equals(guestEmail)
+                && reservation.getStartDate().compareTo(LocalDate.now()) > 0).collect(Collectors.toList());
+
+        view.printReservations(host, reservations);
+
+        if (reservations.size() == 0) {
+
+            return;
+        }
+        int deleteThisid = view.getUserIntInput("Reservation id to be removed: ",
+                reservations.stream().mapToInt(Reservation::getId).min().getAsInt(),
+                reservations.stream().mapToInt(Reservation::getId).max().getAsInt());
+        view.displayReservationDeletionStatus(reservationService.removeReservationById(deleteThisid, host), deleteThisid);
+
+    }
+
+    public void displayMakeReservation() throws DataException {
+        view.displayHeader(MainMenuOption.MAKE_RESERVATION.getMessage());
+
+        String guestEmail = view.getUserStringInput("Guest Email: ");
+        Guest guest = guestService.findGuestByEmail(guestEmail);
+        if (guest == null) {
+            view.guestNotFound(guestEmail);
+            return;
+        }
+
+        String hostEmail = view.getUserStringInput("Host Email: ");
+        Host host = hostService.findHostFromEmail(hostEmail);
+        if (host == null) {
+            view.hostNotFound(hostEmail);
+            return;
+        }
+
+
+        List<Reservation> reservations = reservationService.getReservationForParticularHost(host);
+
+        reservations = reservations.stream().filter(reservation -> reservation.getEndDate().compareTo(LocalDate.now()) >= 0).collect(Collectors.toList());
```

Comment:
You could move this method to the service (i.e. `getFutureReservationForParticularHost()`).